src/libs/ajax/loaded-data/useLoadedData.ts

```diff
@@ -14,13 +14,44 @@ export interface UseLoadedDataArgs<T> {
    *   onError: (errState) => ReportError(errState.error)
    * })
    */
-  onError?: (state: ErrorState<T, unknown>) => void;
+  onError?: (state: ErrorState<S, E>) => void;
+
+  /**
+   * An optional handler that will be called on successful data load.
+   * Note that LoadedData object typing already allows expression of error status, convenient for consumption by
+   * visual components.  This handler is to accommodate additional side effects within the hook consuming
+   * useLoadedData hook
+   * @param state - the ready state as of when the error happened
+   * @example
+   * const [pendingCreate, setPendingCreate] = useLoadedData<true>({
+   *   onError: (readyState) => doSomething(readyState.state)
+   * })
+   */
+  onSuccess?: (state: ReadyState<S>) => void;
 }
 
 /**
  * The Tuple returned by useLoadedData custom helper hook
  */
-export type UseLoadedDataResult<T> = [LoadedState<T, unknown>, (dataCall: () => Promise<T>) => Promise<void>];
+export type UseLoadedDataResult<T, E = unknown> = [LoadedState<T, E>, (dataCall: () => Promise<T>) => Promise<void>];
+
+export type LoadedDataEvents<S, E = unknown> = Pick<UseLoadedDataArgs<S, E>, 'onSuccess' | 'onError'>;
+
+export const useLoadedDataEvents = <S, E = unknown>(loadedData: LoadedState<S, E>, events: LoadedDataEvents<S, E>) => {
```

Comment:
Since there are already multiple modules for various LoadedData related hooks, can we move this to its own module?

Reply:
agreed... I was leaning towards breaking it out myself.